chain/bitcoind_conn.go

```diff
@@ -142,16 +186,36 @@ func NewBitcoindConn(cfg *BitcoindConfig) (*BitcoindConn, error) {
 			"events: %v", err)
 	}
 
-	conn := &BitcoindConn{
-		cfg:           *cfg,
-		client:        client,
-		zmqBlockConn:  zmqBlockConn,
-		zmqTxConn:     zmqTxConn,
-		rescanClients: make(map[uint64]*BitcoindClient),
-		quit:          make(chan struct{}),
+	// Only initialize the PrunedBlockDispatcher when the connected bitcoind
+	// node is pruned.
+	var prunedBlockDispatcher *PrunedBlockDispatcher
+	if chainInfo.Pruned {
+		prunedBlockDispatcher, err = NewPrunedBlockDispatcher(
```

Comment:
Ah ok, safe to ignore my other comment re having extra unused TCP connections sitting around.